sonic-post-syseeprom/scripts/post-syseeprom

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python2
+
+'''
+    post-syseeprom
+    Syseeprom information gathering tool for SONiC
+    This tool will be started during the start phase of pmon container, gathering syseeprom info and write to state DB.
+    It's an one-shot task since syseeprom info are static.
+    With this tool, show syseeprom CLI will be able to get data from state DB instead of access hw or cache.
+'''
+
+try:
+    from sonic_daemon_base.daemon_base import DaemonBase
+    import argparse
+    from swsscommon import swsscommon
+    from sonic_daemon_base import daemon_base
+    from sonic_daemon_base.daemon_base import Logger
+    from sonic_daemon_base.daemon_base import DaemonBase
+except ImportError, e:
+    raise ImportError (str(e) + " - required module not found")
+
+PLATFORM_SPECIFIC_MODULE_NAME = 'eeprom'
+PLATFORM_SPECIFIC_CLASS_NAME = 'board'
+EEPROM_TABLE_NAME = 'EEPROM_INFO'
+
+ERR_ARISTA_PLATFORM = 1
+ERR_FAILED_EEPROM = 2
+ERR_FAILED_UPDATE_DB = 3
+
+SYSLOG_IDENTIFIER = "post-syseeprom"
+
+# Global logger class instance
+logger = Logger(SYSLOG_IDENTIFIER)
+
+class PostSyseeprom(DaemonBase):
+    def __init__(self):
+        DaemonBase.__init__(self)
+
+    def _is_arista_platform(self):
+        (platform_path, hwsku_path) = self.get_path_to_platform_and_hwsku()
+        
+        return 'arista' in platform_path
+
+    def write_info_to_db(self):
+        if self._is_arista_platform():
+            # Arista platform have their own implementation for eeprom tlv parsing, it's not inherited from eeprom base class.
+            # So for now this tool can not support Arista platform, but this will not cause any issue on them. 
+            # decode-syseeprom scripts will cover Arista case, it will still read from HW/cache instead from DB on Arista platform.  
+            logger.log_warning('Arista platform does not support this yet')
+            return ERR_ARISTA_PLATFORM
+
+        eeprom_util = self.load_platform_util(PLATFORM_SPECIFIC_MODULE_NAME, PLATFORM_SPECIFIC_CLASS_NAME)
+        eeprom = eeprom_util.read_eeprom()
+        if eeprom is None :
+            logger.log_error('Failed to read eeprom')
+            return ERR_FAILED_EEPROM
+
+        err = eeprom_util.update_eeprom_db(eeprom)
+        if err:
+            logger.log_error('Failed to update eeprom info to database')
+            return ERR_FAILED_UPDATE_DB
+
+    def clear_db(self):
+        # Same case as in above write_info_to_db() function, Arista platform not supported here.
+        if self._is_arista_platform():
+            logger.log_warning('Arista platform does not support this yet')
+            return ERR_ARISTA_PLATFORM
+
+        state_db = daemon_base.db_connect(swsscommon.STATE_DB)
+        eeprom_tbl = swsscommon.Table(state_db, EEPROM_TABLE_NAME)
+        keys = eeprom_tbl.getKeys()
+        for key in keys:
+            eeprom_tbl._del(key)
+        
+
+def main():
```

Comment:
When inheriting DaemonBase, the main() function should end with a call to my_daemon.run(). This isn't a daemon, and it accepts arguments, so it really doesn't make sense here. I don't think DaemonBase is the appropriate choice here after all.

I think eventually, the Logger class should be broken out of the daemon_base file and placed in a separate package with other common functions all SONiC applications can share (including DaemonBase).

I think you should revert this change, and simply use

```
    from sonic_daemon_base.daemon_base import Logger
```

and

```
SYSLOG_IDENTIFIER = "post-syseeprom"

# Global logger class instance
logger = Logger(SYSLOG_IDENTIFIER)
```

Sorry for the confusion and extra work.

Reply:
agree, changed.

Reply:
> I think eventually, the Logger class should be broken out of the daemon_base file and placed in a separate package with other common functions all SONiC applications can share (including DaemonBase).

But sonic_daemon_base initially was designed in the way to implement DaemonBase.
What's the point in moving DaemonBase from sonic_daemon_base to a separate package eventually?

IMHO, the design of sonic-daemon-base was initially incorrect: DaemonBase should be refactored and equipped only with the mandatory API (signal handlers and run method). The other stuff should be moved to daemon_base directly. Such approach will completely solve all your review comments but requires some extra work to do with aligning the existing stuff of sonic-platform-daemons.

P.S: for logging purposes we can abandon our custom Logger implementation and consider using python embedded mechanisms/libraries.

Reply:
@nazariig: I didn't mention moving the DaemonBase class out of sonic_daemon_base.py, but rather moving the Logger class somewhere where it can be shared, along with other functions that we might want to share between daemons and also one-shot applications. Just a thought to keep in mind as SONiC grows -- how can we reuse code most efficiently.